apps/meteor/tests/unit/server/startup/initialData.tests.ts

```diff
@@ -0,0 +1,235 @@
+import { expect } from 'chai';
+import { beforeEach, it } from 'mocha';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+const getUsersInRole = sinon.stub();
+const checkUsernameAvailability = sinon.stub();
+const validateEmail = sinon.stub();
+const addUserRolesAsync = sinon.stub();
+const models = {
+	Settings: {},
+	Rooms: {},
+	Users: {
+		create: sinon.stub(),
+		findOneByEmailAddress: sinon.stub(),
+	},
+};
+const setPasswordAsync = sinon.stub();
+const settingsGet = sinon.stub();
+
+const { insertAdminUserFromEnv } = proxyquire.noCallThru().load('../../../../server/startup/initialData.js', {
+	'meteor/accounts-base': {
+		Accounts: {
+			setPasswordAsync,
+		},
+	},
+	'meteor/meteor': {
+		Meteor: {
+			startup: sinon.stub(),
+		},
+	},
+	'../../app/authorization/server': {
+		getUsersInRole,
+	},
+	'../../app/file-upload/server': {},
+	'../../app/file/server': {},
+	'../../app/lib/server/functions/addUserToDefaultChannels': {},
+	'../../app/lib/server/functions/checkUsernameAvailability': {
+		checkUsernameAvailability,
+	},
+	'../../app/settings/server': {
+		settings: { get: settingsGet },
+	},
+	'../../lib/emailValidator': {
+		validateEmail,
+	},
+	'../lib/roles/addUserRoles': {
+		addUserRolesAsync,
+	},
+	'@rocket.chat/models': models,
+});
+
+describe('insertAdminUserFromEnv', () => {
+	beforeEach(() => {
+		getUsersInRole.reset();
+		checkUsernameAvailability.reset();
+		validateEmail.reset();
+		addUserRolesAsync.reset();
+		models.Users.create.reset();
+		models.Users.findOneByEmailAddress.reset();
+		setPasswordAsync.reset();
+		settingsGet.reset();
+		process.env.ADMIN_PASS = 'pass';
+	});
+
+	it('should do nothing if process.env.ADMIN_PASS is empty', async () => {
+		process.env.ADMIN_PASS = '';
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.false;
+		expect(result).to.be.undefined;
+	});
+	it('should do nothing if theres already an admin user', async () => {
+		getUsersInRole.returns({ count: () => 1 });
+
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.true;
+		expect(validateEmail.called).to.be.false;
```

Comment:
Instead of asserting if this function is called since it's inside another condition, it'd be better to test if the user was created or not (`Users.create()`). WDYT?

Reply:
There's some tests for that case below. These ones is just to check that both these functions were not called when the env is not set.
(Actually when env is not set it shouldn't call anything, so i just picked these 2 from inside the if)